File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        final=""
        res=0
        x=0
        if(len(s)==1):
            return 1
        while(x<len(s)):
            if s[x] not in final:
                final+=s[x]
                x+=1
                continue 
            else:
                if res<len(final):
                    res=len(final) 
                # print(final)
                idx=final.index(s[x])
                if(idx<len(final)-1):
                    final=final[idx+1:] 
                else:
                    final=""
        # print(final)
        if(len(final)>res):
            res=len(final)
            
        return res
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i=0;
        int n= s.length(); 
        int res= 0;
        // cout<<n;
        while(i<n){
            int j=i+1;
            unordered_set<char> st; 
            st.insert(s[i]); 
            
            while(j<n){
                if(st.find(s[j])==st.end()) { 
                    st.insert(s[j]); 
                    j+=1;
                }
                else{
                    res=max(res, j-i);
                    break;
                }
            }
            
            
            res=max(res,j-i);
            i+=1;
        }
        return res;
    }
        
};